### PR TITLE
Add gLogging for debugging

### PR DESCRIPTION
**Pull request by gdbelvin**
There are the logging statements I added while debugging the merkle tree. 
These may be useful for others in the future.  They're relatively verbose, so they're hidden behind relatively high verbose glog numbers.

### CODE REVIEW

File: storage/cache/subtree_cache.go [hunk 1]
```diff
@@ -246,6 +251,7 @@ func (s *SubtreeCache) getNodeHashUnderLock(id storage.NodeID, getSubtree GetSub
 	prefixKey := string(px)
 	c := s.subtrees[prefixKey]
 	if c == nil {
+		glog.Warningf("Cache miss for %x so we'll try to fetch from storage", prefixKey)
```
Comment (AlCutter) [inline storage/cache/subtree_cache.go]: Probably don't want a Warning here - that typically flushes to disk (or at least it does in the C++ glog)

File: storage/mysql/tree_storage.go [hunk 2]
```diff
@@ -252,6 +264,16 @@ func (t *treeTX) getSubtrees(ctx context.Context, treeRevision int64, nodeIDs []
 }
 
 func (t *treeTX) storeSubtrees(ctx context.Context, subtrees []*storagepb.SubtreeProto) error {
+	if glog.V(4) {
+		glog.Infof("storeSubtrees(")
+		for _, s := range subtrees {
+			glog.V(4).Infof("  prefix: %x, depth: %d", s.Prefix, s.Depth)
```
Comment (AlCutter) [inline storage/mysql/tree_storage.go]: Elsewhere you've removed the `V(x)` bit if you're inside an `if` block which already checked that

File: storage/mysql/tree_storage.go [hunk 3]
```diff
@@ -244,6 +247,15 @@ func (t *treeTX) getSubtrees(ctx context.Context, treeRevision int64, nodeIDs []
 			subtree.Prefix = []byte{}
 		}
 		ret = append(ret, &subtree)
+
+		if glog.V(4) {
+			glog.Infof("  subtree: NID: %x, prefix: %x, depth: %d",
+				subtreeIDBytes, subtree.Prefix, subtree.Depth)
+			for k, v := range subtree.Leaves {
+				b, _ := base64.StdEncoding.DecodeString(k)
```
Comment (AlCutter) [inline storage/mysql/tree_storage.go]: If you've got debugging logging on at this point, you'd probably want to know if one of your key string is invalid base64 vs an empty string - that could be _days_ of hair pulling :)

File: storage/mysql/tree_storage.go [hunk 4]
```diff
@@ -252,6 +264,16 @@ func (t *treeTX) getSubtrees(ctx context.Context, treeRevision int64, nodeIDs []
 }
 
 func (t *treeTX) storeSubtrees(ctx context.Context, subtrees []*storagepb.SubtreeProto) error {
+	if glog.V(4) {
+		glog.Infof("storeSubtrees(")
+		for _, s := range subtrees {
+			glog.V(4).Infof("  prefix: %x, depth: %d", s.Prefix, s.Depth)
+			for k, v := range s.Leaves {
+				b, _ := base64.StdEncoding.DecodeString(k)
```
Comment (AlCutter) [inline storage/mysql/tree_storage.go]: same invalid b64 comment here too

File: storage/cache/subtree_cache.go [hunk 5]
```diff
@@ -284,6 +290,10 @@ func (s *SubtreeCache) getNodeHashUnderLock(id storage.NodeID, getSubtree GetSub
 	} else {
 		nh = c.InternalNodes[sfxKey]
 	}
+	if glog.V(4) {
+		b, _ := base64.StdEncoding.DecodeString(sfxKey)
```
Comment (AlCutter) [inline storage/cache/subtree_cache.go]: invalid b64 checking here too

File: storage/cache/subtree_cache.go (hunk 1 above)
Comment (gdbelvin) [inline storage/cache/subtree_cache.go]: moved to `glog.Infof`

File: storage/cache/subtree_cache.go (hunk 5 above)
Comment (gdbelvin) [inline storage/cache/subtree_cache.go]: done

File: storage/mysql/tree_storage.go (hunk 3 above)
Comment (gdbelvin) [inline storage/mysql/tree_storage.go]: done

File: storage/mysql/tree_storage.go (hunk 2 above)
Comment (gdbelvin) [inline storage/mysql/tree_storage.go]: done

File: storage/mysql/tree_storage.go (hunk 4 above)
Comment (gdbelvin) [inline storage/mysql/tree_storage.go]: done

**Comment by gdbelvin** [comment]
Updated. PTAL

**Pull request merged**